tools/c7n_oci/c7n_oci/output.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import os
+
+import oci
+from oci.exceptions import ServiceError
+
+from c7n.output import blob_outputs, BlobOutput, log_outputs, LogOutput
+from c7n_oci.constants import PROFILE, OCI_LOG_COMPARTMENT_ID
+from c7n_oci.log import OCILogHandler
+from c7n_oci.session import SessionFactory
+
+
+@blob_outputs.register("oci")
+class OCIObjectStorageOutput(BlobOutput):
+    log = logging.getLogger('custodian.oci.output.OCIObjectStorageOutput')
+
+    def __init__(self, ctx, config):
+        super(OCIObjectStorageOutput, self).__init__(ctx, config)
+        if PROFILE in self.config.keys():
+            session = SessionFactory(profile=self.config.profile)()
+        else:
+            session = ctx.session_factory()
+        self.os_client = session.client("oci.object_storage.ObjectStorageClient")
+        self.namespace = self.os_client.get_namespace().data
+        self.create_bucket_if_does_not_exist()
```

Comment:
fwiw, as a general rule we prefer not to do api calls in class __init__, at least wrt to filters/actions/resources as we treat them as fly weights when expanding across regions, or validating, filtering selection for execution.

Reply:
Got it! Removed api calls from the init method for both object storage and logging